scripts/extract-codeblock.awk

```diff
@@ -0,0 +1,23 @@
+BEGIN {CODEBLOCK = 0; SNIPPET = 0; RED = "\\033[0;31m"; NO_COLOUR = "\\033[0m"; }
+BEGINFILE {
+    if (ERRNO != "") {
+        print "couldn't open " FILENAME " skipping"
+        nextfile
+    }
+}
+/^\[source,yaml\]/                           { SNIPPET = FNR; next} # code block follows between `----`
+SNIPPET && !CODEBLOCK && /[-]{4}/            { CODEBLOCK = 1; COUNTER = COUNTER++ ; next} # code block found
+CODEBLOCK && /[-]{4}/                        {
+    yq_cmd = ("echo '" buf "' | yq 'true' > /dev/null")
```

Comment:
would be cool to inject the validation command here as well so it doesn't just assume `yaml` with `yq`